cmd/get/cloudconnections/cloudconnections.go

```diff
@@ -0,0 +1,137 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cloudconnections
+
+import (
+	"os"
+	"strings"
+
+	"github.com/IBM-Cloud/power-go-client/ibmpisession"
+	"github.com/IBM/go-sdk-core/v5/core"
+	"github.com/spf13/cobra"
+	"k8s.io/klog/v2"
+
+	"github.com/ppc64le-cloud/pvsadm/pkg"
+	"github.com/ppc64le-cloud/pvsadm/pkg/client"
+	"github.com/ppc64le-cloud/pvsadm/pkg/utils"
+)
+
+// Struct to contain the information related to a cloud connection.
+type cloudConnectionDetails struct {
+	name, cloudConnId, state, zone string
+	workspaces                     map[string]struct{}
+}
+
+// Struct to contain the information related to a workspace.
+type workspaceDetails struct {
+	name, guid string
+}
+
+// Map to contain the zone and their associated workspaces it contains at an account level.
+var zoneWorkspaces = make(map[string][]workspaceDetails)
+
+var Cmd = &cobra.Command{
+	Use:   "cloud-connections",
+	Short: "List the existing cloud connections in the account",
+	Long:  "List the existing cloud connections enabled across all workspaces in the account",
+	PreRunE: func(md *cobra.Command, args []string) error {
+		opt := pkg.Options
+		// TODO: The GetAuthenticatorFromEnvironment function seems to refer to "IBMCLOUD_APIKEY" rather than IBMCLOUD_API_KEY as used in the project.
+		// In order to use the resourcecontrollerv2's functionality, the IBMCLOUD_API_KEY is re-exported as IBMCLOUD_APIKEY.
+		os.Setenv("IBMCLOUD_APIKEY", opt.APIKey)
+		return nil
+	},
+	RunE: func(cmd *cobra.Command, args []string) error {
+		opt := pkg.Options
+		c, err := client.NewClientWithEnv(opt.APIKey, opt.Environment, opt.Debug)
+		if err != nil {
+			klog.Errorf("failed to create a session with IBM cloud: %v", err)
+			return err
+		}
+
+		// Retrieve all workspaces that are available in the account.
+		workspaceInstances, err := c.ListWorkspaceInstances()
+		if err != nil {
+			return err
+		}
+		for _, workspaceInstance := range workspaceInstances.Resources {
+			if _, exists := zoneWorkspaces[*workspaceInstance.RegionID]; !exists {
+				zoneWorkspaces[*workspaceInstance.RegionID] = make([]workspaceDetails, 0)
+			}
+			zoneWorkspaces[*workspaceInstance.RegionID] = append(zoneWorkspaces[*workspaceInstance.RegionID], workspaceDetails{name: *workspaceInstance.Name, guid: *workspaceInstance.GUID})
+		}
+		cloudConnections := map[string]cloudConnectionDetails{}
+		klog.Info("Listing cloud connections across all workspaces, please wait..")
+		// Create a IBM PI Session per zone and reuse them across the workspaces in the same zone.
+		for workspaceZone, workspaces := range zoneWorkspaces {
+			authenticator := &core.IamAuthenticator{ApiKey: c.Config.BluemixAPIKey, URL: *c.Config.TokenProviderEndpoint}
```

Comment:
can we move this outside of the for loop to reuse?

Reply:
Made the required change.
Thanks!